src/v/security/gssapi_authenticator.cc

```diff
@@ -0,0 +1,514 @@
+        return errc::invalid_scram_state;
+    }
+    vlog(
+      seclog.trace,
+      "gss {} gss_export_cred: {}",
+      _state,
+      hex(bytes_view{token}));
+
+    gss::buffer display_name;
+    major_status = ::gss_display_name(
+      &minor_status, server_name, &display_name, &_doid);
+    if (major_status != GSS_S_COMPLETE) {
+        display_status("gss server name", major_status, minor_status);
+    } else {
+        vlog(
+          seclog.trace,
+          "gss {} server name {}:{} = {}",
+          _state,
+          major_status,
+          minor_status,
+          std::string_view{display_name});
+    }
+    _state = state::more;
+    return outcome::success();
+}
+
+result<bytes> gssapi_authenticator::more(bytes_view auth_bytes) {
+    gss::buffer_view recv_tok{auth_bytes};
+    gss::buffer send_tok;
+
+    vlog(
+      seclog.trace,
+      "recv_tok: length: {} - {}",
+      auth_bytes.length(),
+      hex{auth_bytes});
+
+    if (auth_bytes.length() == 0) {
+        if (auto res = finish(); res.has_error()) {
+            return res.assume_error();
+        }
+        return bytes{};
+    }
+
+    OM_uint32 major_status{}, minor_status{}, ret_flags{};
+
+    major_status = ::gss_accept_sec_context(
+      &minor_status,
+      &_context,
+      _server_creds,
+      &recv_tok,
+      GSS_C_NO_CHANNEL_BINDINGS,
+      &_client_name,
+      &_doid,
+      &send_tok,
+      &ret_flags,
+      NULL,  /* ignore time_rec */
+      NULL); /* ignore del_cred_handle */
+
+    vlog(
+      seclog.trace,
+      "gss {} acccept sec context {}:{}",
+      _state,
+      major_status,
+      minor_status);
+    if (
+      major_status != GSS_S_COMPLETE && major_status != GSS_S_CONTINUE_NEEDED) {
+        display_status("gss doid", major_status, minor_status);
+        _state = state::failed;
+        return errc::invalid_scram_state;
+    }
+
+    bytes ret{bytes_view{send_tok}};
+    if (major_status == GSS_S_COMPLETE) {
+        _state = state::ssfcap;
+        if (auto res = check(); res.has_error()) {
+            return res.assume_error();
+        }
+    } else if (major_status == GSS_S_CONTINUE_NEEDED) {
+        _state = state::more;
+    }
+    vlog(seclog.trace, "gss {} sending: {}", _state, hex(ret));
+    vlog(
+      seclog.trace, "gss {} flags: {}", _state, display_ctx_flags(ret_flags));
+
+    return ret;
+}
+
+result<bytes> gssapi_authenticator::ssfcap(bytes_view auth_bytes) {
+    if (!auth_bytes.empty()) {
+        _state = state::failed;
+        vlog(
+          seclog.info,
+          "gss {} expected empty response: {}",
+          _state,
+          hex(auth_bytes));
+        return errc::invalid_credentials;
+    }
+
+    // OM_uint32 maj_stat = 0, min_stat = 0;
+    // gss_buffer_set_t bufset = GSS_C_NO_BUFFER_SET;
+    // gss_OID ssf_oid = ::GSS_C_SEC_CONTEXT_SASL_SSF;
+    // uint32_t ssf;
+
+    // maj_stat = gss_inquire_sec_context_by_oid(
+    //   &min_stat, _context, ssf_oid, &bufset);
```

Comment:
what's our policy on commented out code?  In the past I've flag it to be deleted or putting a comment around it indicating why it's sticking around.

Reply:
I'll clean this up

Reply:
Done